libraries/hotstuff/qc_chain.cpp

```diff
@@ -630,6 +642,8 @@ namespace eosio::hotstuff {
    // Called from the main application thread
    void qc_chain::on_beat(){
 
+      _sender_connection_id = std::nullopt;
```

Comment:
Storing the sender connection id seems like an anti-pattern to me. It provides a point of contention between the threads interacting with `qc_chain`. I know this is all protected by a large mutex, but if we want to remove that in the future seems much cleaner to pass along connection id via the call stack.